jbmc/src/java_bytecode/README.md

```diff
@@ -43,6 +43,45 @@ To be documented.
 
 To be documented.
 
+\section java-bytecode-remove-instanceof Remove instanceof
+
+Java has an operator called `instanceof`, which is used to check if an instance of a class is of a particular type. For example given that `A` and `B` are the identifiers of classes, which derive directly from Object. If the following example java code is executed -
+```java
+Object a = new A();
+boolean x = a instanceof A; // true
+boolean y = a instanceof B; // false
+```
+Then, `x` will be set to `true` and `y` will be set to `false`.
+
+`instanceof` also works with inheritance. Given classes `C` and `D`, where `C` derrives from `Object` and `D` derives from `C`. If the following example java code is executed -
+```java
+Object c = new C();
+Object d = new D();
+boolean b1 = a instanceof A; // true
+boolean b2 = a instanceof B; // false
+boolean b3 = b instanceof A; // true
+boolean b4 = b instanceof B; // true
+```
+Then, `b1` will be `true`, `b2` will be `false`, `b3` will be `true` and `b4` will be `true`.
+
+`instanceof` has its own instruction type in the java bytecode. We need to rewrite it in terms of other, more fundamental operations in order to analyse it. This operation is referred to as "lowering" or in this specific case - `remove_instanceof`.
```

Comment:
One think that I think is useful is specific Java -> Java bytecode that would be translated by this step. I'd also avoid redefining "lowering" and why it is needed since imagine you are reading about all these different transformations, you don't want to keep reading why lowering is necessary. 

I also don't know, but apart from the `instanceof` operator, does any other Java code get translated into an `instanceof` (e.g. do casts? catch blocks?)

Reply:
OK I will add that overview sentence.

Reply:
I have taken this overview sentence and made it into this PR: #2792